From a6bd3166fabdbe26c90ea96f36797b2be3da9bef Mon Sep 17 00:00:00 2001
From: Sam Ravnborg <srn@skov.dk>
Date: Fri, 17 Feb 2017 10:31:03 +0100
Subject: [PATCH] board: fix build warning with ARM9-CPU

ARM9-CPU shall not (for unknown reasons) call halt()
But lets prints "failed", and as a side effect we no longer
have a build warning
---
 arch/arm/lib/board.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/lib/board.c b/arch/arm/lib/board.c
index 8b0decb..4c70884 100644
--- a/arch/arm/lib/board.c
+++ b/arch/arm/lib/board.c
@@ -526,8 +526,9 @@ void board_init_r(gd_t *id, ulong dest_addr)
 # endif /* CONFIG_SYS_FLASH_CHECKSUM */
 	} else {
 	//
-#if !defined(CONFIG_ARM9_CPU)
 		puts(failed);
+
+#if !defined(CONFIG_ARM9_CPU)
 		hang();
 #endif /* CONFIG_ARM9_CPU */
 	}
-- 
1.8.3.1

